src/lib/dyn_array/dyn_array.ml

```diff
@@ -1,3 +1,4 @@
+open Extlib
 open Core
```

Comment:
Can I request that we use `Core_kernel` here, and in the `dune` file? I don't think full `Core` is needed.

Reply:
Both `extlib` and `batteries` have a `DynArray`. The `dune` file suggests we're using `batteries`, while the `.ml` file suggests we're using `extlib`. 

Maybe I'm just confused ...